src/__tests__/requirement06.test.js

```diff
@@ -1,25 +1,21 @@
 import React from 'react';
 import { render, screen, waitFor, fireEvent } from '@testing-library/react';
 import App from '../App';
-import * as api from '../services/api';
-import mockedCategoriesResult from '../__mocks__/categories';
 import mockedQueryResult from '../__mocks__/query';
-
-jest.mock('../services/api');
-api.getCategories.mockImplementation(
-  () => Promise.resolve(mockedCategoriesResult),
-);
-api.getProductsFromCategoryAndQuery.mockImplementation(
-  () => Promise.resolve(mockedQueryResult),
-);
+import mockFetch from '../__mocks__/mockFetch';
 
 describe('6 - Selecione uma categoria e mostre somente os produtos daquela categoria', () => {
   it(`Filtra corretamente os produtos de uma página para exibir somente os daquela
       categoria`, async () => {
+
+    jest.spyOn(global, 'fetch').mockImplementation((url) => {
+      return mockFetch(url);
+    });
     render(<App />);
-    await waitFor(() => expect(api.getCategories).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
     fireEvent.click(screen.getAllByTestId('category')[0]);
-    await waitFor(() => expect(api.getProductsFromCategoryAndQuery).toHaveBeenCalled());
+
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
```

Comment:
Nesse caso, se a gente quiser garantir que o `fetch` foi chamado duas vezes, é melhor usar o método `toHaveBeenCalledTimes(2)`.
Como o método `toHaveBeenCalled` verifica que se a função foi chamada 1 ou mais vezes (>=1), aqui nessa linha sempre vai dar `true`, porque o `fetch` já foi chamado uma vez antes.
```suggestion
    await waitFor(() => expect(global.fetch).toHaveBeenCalledTimes(2));
```